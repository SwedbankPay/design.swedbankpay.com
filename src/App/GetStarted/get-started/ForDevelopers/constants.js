import identityRoute from "@src/App/routes/identity";
import componentRoute from "@src/App/routes/components";

const iconography = identityRoute[0].routes.find(
	(component) => component.title === "Iconography"
);

export const browsers = [
	{
		url: "img/documentation/forDevelopers/chrome.png",
		alt: "Chrome logo",
		text: "Yes",
	},
	{
		url: "img/documentation/forDevelopers/edge.png",
		alt: "Edge logo",
		text: "Yes",
	},
	{
		url: "img/documentation/forDevelopers/firefox.png",
		alt: "Firefox logo",
		text: "Yes",
	},
	{
		url: "img/documentation/forDevelopers/opera.png",
		alt: "Opera logo",
		text: "Yes",
	},
	{
		url: "img/documentation/forDevelopers/safari.png",
		alt: "Safari logo",
		text: "Yes",
	},
];

export const gridAndBreakpoints = [
	{
		size: "Mobile",
		breakpoints: "320-767 px",
		columns: "4",
		margins: "16 px",
	},
	{
		size: "Tablet",
		breakpoints: "767-1023 px",
		columns: "8",
		margins: "24 px",
	},
	{
		size: "Small desktop",
		breakpoints: "1024-1439 px",
		columns: "12",
		margins: "24 px",
	},
	{
		size: "Large desktop",
		breakpoints: "â‰¥ 1440 px",
		columns: "12",
		margins: "24 px",
	},
];

export const discoverMore = [
	{
		title: componentRoute[0].title,
		url: componentRoute[0].path,
		text: "See all available components",
	},
	{
		title: iconography.title,
		url: iconography.path,
		text: "Our icon library and how to use it",
	},
];
