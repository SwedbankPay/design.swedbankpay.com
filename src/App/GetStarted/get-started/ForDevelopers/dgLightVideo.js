import React from "react";

const DGLightVideo = () => (
	<div className="video">
		<iframe
			frameBorder="0"
			allowFullScreen
			className="responsive-iframe"
			src={
				"https://player.vimeo.com/video/727352249?h=232b0bec4c&amp;badge=0&amp;autopause=0&amp;player_id=0&amp;app_id=58479"
			}
		/>
	</div>
);

export default DGLightVideo;
