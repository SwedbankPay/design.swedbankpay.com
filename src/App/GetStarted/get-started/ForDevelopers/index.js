import React from "react";
import { Link } from "react-router-dom";
import { browsers, gridAndBreakpoints, discoverMore } from "./constants";

import { ComponentPreview, DocContainer } from "@docutils";
import CodeTags from "@components/CodeTags";

const basename = process.env.basename;
const brand = process.env.brand;
const brandTitle = process.env.brandTitle;

const Installation = () => (
    <section>
        <h2 id="installation">Installation with CDN</h2>
        <p>To quickly add the {brandTitle} Design Guide to your project, include the snippets below in your project.</p>
        <h3>Include in the header</h3>
        <p>Copy-paste the following CSS code into <CodeTags type="secondary" code="<head>"/> before all the other stylesheets in order to load our CSS.</p>
        <ComponentPreview language="html" codeFigure>
            <link rel="stylesheet" href={`${window.location.origin + basename}styles/${brand}.css`} />
        </ComponentPreview>

        <h3>Include at the bottom</h3>
        <p>Many of our components requires the use of JavaScript in order to function. Place the following <CodeTags type="secondary" code="<Script>"/> tag near the end of your pages, right before the closing <CodeTags type="secondary" code="</body>"/> tag to enable them.</p>
        <ComponentPreview language="html" codeFigure>
            <script src={`${window.location.origin + basename}scripts/dg.js`} global="true" autoload="true"></script>
        </ComponentPreview>
        <p>The <CodeTags type="secondary" code="global"/> attribute adds the script to the global scope. The <CodeTags type="secondary" code="autoload"/> attribute automatically initiates all components when loaded. Great for plug and play.</p>
        <h2 id="installation-npm">Installation with NPM</h2>
        <p>If you want to host the Design Guide files in your own app, run the following command.</p>
        <ComponentPreview language="terminal" codeFigure>
            npm install @swedbankpay/design-guide
        </ComponentPreview>
        <h3>Usage with NPM</h3>
        <p>When importing the Design Guide javascript, you can choose to either import the whole thing or only the components you want to use.</p>
        <ComponentPreview language="javascript" codeFigure>
            import dg from &quot;@swedbankpay/design-guide&quot;; {"\n"}
            dg.script.initAll();
        </ComponentPreview>
        <ComponentPreview language="javascript" codeFigure>
            import &#123; sheet &#125; from &quot;@swedbankpay/design-guide&quot;;  {"\n"}
            sheet.init();
        </ComponentPreview>
        <p>For our CSS you can choose to import it from a <CodeTags type="secondary" code="css/less/scss"/> file, from a <CodeTags type="secondary" code="js"/> file if your bundler supports it, or point to the file directly from your bundler.</p>
        <p>Example from a <CodeTags type="secondary" code=".less"/> file</p>
        <ComponentPreview language="css" codeFigure>
            @import url(&quot;@swedbankpay/design-guide/src/less/{brand}.less&quot;);
        </ComponentPreview>
        <p>Example from <CodeTags type="secondary" code="angular.json"/></p>
        <ComponentPreview language="json" codeFigure>
        &quot;styles&quot;: [ {"\n"}
            &nbsp; &quot;src/app.less&quot;, {"\n"}
            &nbsp; &quot;./node_modules/@swedbankpay/design-guide/src/less/{brand}.less&quot; {"\n"}
        ],
        </ComponentPreview>
        <p>Example from <CodeTags type="secondary" code="nuxt.config.js"/></p>
        <ComponentPreview language="javascript" codeFigure>
        &quot;css&quot;: [ {"\n"}
            &nbsp; &quot;./node_modules/@swedbankpay/design-guide/src/less/{brand}.less&quot; {"\n"}
        ],
        </ComponentPreview>
        <h3>How to initialize our JavaScript components</h3>
        <p>Our script runs <CodeTags type="secondary" code="dg.script.initAll();"/> on DOMContentLoaded making it so you have to run <CodeTags type="secondary" code="dg.<component_name>.init();"/> manually when a component is rendered after the event has fired.</p>

        <p>To initialize components use <CodeTags type="secondary" code="dg.<component_name>.init();"/>. If the function is called without arguments it will target all components with class <CodeTags type="secondary" code="<component_name>"/> on the page. Initializing only one component can be done by passing an ID, it must match an existing components ID and will print a warning message if the value is invalid.</p>

        <p>If only one component is initialized then the init function will return one object. If several components are initialized it will return an array of objects.</p>
        <ComponentPreview language="html" codeFigure removeOuterTag>
                dg.component.init();{"\n"}
                dg.component.init(&quot;demo-component-id&quot;);
        </ComponentPreview>

        <h3>Browser Support</h3>
        <p>The design guide is tested and currently supported by these browsers.</p>
        <div className="browser-support">
            {browsers.map(browser => (
                <div key={browser.url}>
                    <img src={basename + browser.url} alt={browser.alt}/>
                    <p>{browser.text}</p>
                </div>
            ))}
        </div>
        <h4>Browser support for products using the Design Guide</h4>
        <p>It is up to every Project manager and development team to decide which browsers their product should support. If you want to support other browsers than the ones supported by the Design Guide you need to maintain and test the component code in in your project. </p>
    </section>
);
const GridAndBreakpoints = () => (
    <section>
        <h2 id="grid-and-breakpoints">Grid and breakpoints</h2>
        <p>Our grid system is a customized version of the bootstrap grid and functions almost the same, with most of the Bootstrap grid system classes available. To learn more about how to use <Link to="/identity/grid">our 12 column grid system.</Link></p>
        <table className="table table-plain">
            <thead>
                <tr>
                    <th>Size</th>
                    <th>Breakpoint Range</th>
                    <th>Columns</th>
                    <th>Margins & Gutter</th>
                </tr>
            </thead>
            <tbody>
                {gridAndBreakpoints.map(element => (
                    <tr key={element.size}>
                        <td>{element.size}</td>
                        <td>{element.breakpoints}</td>
                        <td>{element.columns}</td>
                        <td>{element.margins}</td>
                    </tr>
                ))}
            </tbody>
        </table>

        <h3>8-point spacing system</h3>
        <p>In addition to the Grid, we also use a 8-point spacing system for spacing between different UI elements. In practice this means we use multiples of 8 in order to define dimensions, padding and margins between the elements. Read more about how it is used in our <Link to="/identity/spacing">Spacing section</Link>.</p>
    </section>
);
const Contributing = () => (
    <section>
        <h2 id="contributing">Contributing</h2>
        <p>Our project is fully open source and we would love to have you contribute to make the design guide better! You can create a ticket directly in <a href="https://github.com/SwedbankPay/design.swedbankpay.com" className="icon-link m-0" target="_blank" rel="noopener noreferrer"><span className="mr-2">our Git Repository</span><i className="material-icons" aria-hidden="true">open_in_new</i></a> or create a feature request here.</p>
        <p>Please make sure to check out our <a href="https://github.com/SwedbankPay/design.swedbankpay.com#contributing" className="icon-link m-0" target="_blank" rel="noopener noreferrer"><span className="mr-2">Contribution Guide</span><i className="material-icons" aria-hidden="true">open_in_new</i></a>, it includes guidelines for how to contribute with new components and develop for the project.</p>
    </section>
);
const Feedback = () => (
    <section>
        <h2 id="feedback">Feedback</h2>
        <p>We want to do our best to provide top quality components for you to use, but this is only possible with the help from you! If you have any suggestions for how we could potentially improve a component, please donâ€™t hesitate to reach out, you can find us in the <strong>#design-guide-feedback</strong> Slack channel.</p>
    </section>
);
const DiscoverMore = () => (
    <section>
        <h2 id="discover-more">Discover more</h2>
        <div className="component-overview hide-arrow-icon">
            {discoverMore.map(card => (
                <Link key={card.url} to={card.url} className="cards cards-primary cards-wide">
                    <div className="cards-content m-0">
                        <span className="h3">{card.title}</span><span>{card.text}</span>
                    </div>
                    <i className="material-icons">arrow_forward</i>
                </Link>
            ))}
        </div>
    </section>
);

const ForDevelopers = () => (
    <DocContainer>
        <p className="lead mb-5">To get you up and running with the {brandTitle} Design Guide as quickly as possible, we have prepared this section showing you how to install it and what you will need to know in order to contribute to the project!</p>
        <Installation/>
        <GridAndBreakpoints />
        <Contributing />
        <Feedback />
        <DiscoverMore />
    </DocContainer>
);

export default ForDevelopers;

export { Installation, GridAndBreakpoints, Contributing, Feedback, DiscoverMore };
