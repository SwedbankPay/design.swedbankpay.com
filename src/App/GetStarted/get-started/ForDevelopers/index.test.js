import React from "react";
import renderer from "react-test-renderer";

import ForDevelopers, { Installation, GridAndBreakpoints, Contributing, Feedback, DiscoverMore, DGLightVideo } from "./index";

describe("GetStarted: ForDevelopers", () => {
    it("is defined", () => {
        expect(ForDevelopers).toBeDefined();
    });

    describe("Installation", () => {
        it("is defined", () => {
            expect(Installation).toBeDefined();
        });

        it("renders", () => {
            const wrapper = renderer.create(<Installation />);

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });

    describe("GridAndBreakpoints", () => {
        it("is defined", () => {
            expect(GridAndBreakpoints).toBeDefined();
        });
    });

    describe("Contributing", () => {
        it("is defined", () => {
            expect(Contributing).toBeDefined();
        });

        it("renders", () => {
            const wrapper = renderer.create(<Contributing />);

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });

    describe("Feedback", () => {
        it("is defined", () => {
            expect(Feedback).toBeDefined();
        });

        it("renders", () => {
            const wrapper = renderer.create(<Feedback />);

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });

    describe("DiscoverMore", () => {
        it("is defined", () => {
            expect(DiscoverMore).toBeDefined();
        });
    });

    describe("DGLightVideo", () => {
        it("is defined", () => {
            expect(DGLightVideo).toBeDefined();
        });

        it("renders", () => {
            const wrapper = renderer.create(<DGLightVideo />);

            expect(wrapper.toJSON()).toMatchSnapshot();
        });
    });
});
