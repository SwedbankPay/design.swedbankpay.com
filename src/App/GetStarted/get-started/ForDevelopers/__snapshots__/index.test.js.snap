// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStarted: ForDevelopers Contributing renders 1`] = `
<section>
  <h2
    id="contributing"
  >
    Contributing
  </h2>
  <p>
    Our project is fully open source and we would love to have you contribute to make the design guide better! You can create a ticket directly in
     
    <a
      className="icon-link m-0"
      href="https://github.com/SwedbankPay/design.swedbankpay.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="mr-2"
      >
        our Git Repository
      </span>
      <i
        aria-hidden="true"
        className="at-arrow-up-from-square small align-self-baseline"
      />
    </a>
     
    or create a feature request here.
  </p>
  <p>
    Please make sure to check out our
     
    <a
      className="icon-link m-0"
      href="https://github.com/SwedbankPay/design.swedbankpay.com#contributing"
      rel="noopener noreferrer"
      target="_blank"
    >
      <span
        className="mr-2"
      >
        Contribution Guide
      </span>
      <i
        aria-hidden="true"
        className="at-arrow-up-from-square small align-self-baseline"
      />
    </a>
    , it includes guidelines for how to contribute with new components and develop for the project.
  </p>
</section>
`;

exports[`GetStarted: ForDevelopers DGLightVideo renders 1`] = `
<div
  className="video"
>
  <iframe
    allowFullScreen={true}
    className="responsive-iframe"
    frameBorder="0"
    src="https://player.vimeo.com/video/727352249?h=232b0bec4c&amp;badge=0&amp;autopause=0&amp;player_id=0&amp;app_id=58479"
  />
</div>
`;

exports[`GetStarted: ForDevelopers Feedback renders 1`] = `
<section>
  <h2
    id="feedback"
  >
    Feedback
  </h2>
  <p>
    We want to do our best to provide top quality components for you to use, but this is only possible with the help from you! If you have any suggestions for how we could potentially improve a component, please donâ€™t hesitate to reach out, you can find us in the
     
    <strong>
      #design-guide-feedback
    </strong>
     Slack channel.
  </p>
</section>
`;

exports[`GetStarted: ForDevelopers Installation renders 1`] = `
<section>
  <section>
    <h2
      id="installation-npm"
    >
      Installation with NPM
    </h2>
    <p>
      If you want to host the Design Guide files in your own app, run the following command.
    </p>
    <figure>
      <div
        className="code-view"
      >
        <header
          className="code-view-header"
        >
          TERMINAL
          <button
            aria-describedby="tooltipCopy"
            className="tooltip"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="at-clipboard small"
            />
            <div
              id="tooltipCopy"
              role="tooltip"
            >
              Copy to clipboard
            </div>
          </button>
        </header>
        <table
          className="code-view-table"
        >
          <tbody>
            <tr>
              <td
                className="code-view-numbers"
              >
                <pre>
                  1
                  

                </pre>
              </td>
              <td
                className="code-view-code"
              >
                <pre>
                  npm install @swedbankpay/design-guide
                  

                </pre>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </figure>
    <section>
      <h3>
        Import methods comparisons
      </h3>
      <p>
        The Design-System NPM package offers 2 possibilities:
        <ul>
          <li>
            original unprocessed source files
            <ul>
              <li>
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/src/...
                </code>
              </li>
              <li>
                <code
                  className="code-tags code-tags-secondary"
                >
                  .js
                </code>
                 &
                 
                <code
                  className="code-tags code-tags-secondary"
                >
                  .less
                </code>
              </li>
              <li>
                individual components files + globals, utils & variables
              </li>
              <li>
                unprocessed for browser support, each project is responsible for ensuring the browser support they need & processing the style files from 
                <code
                  className="code-tags code-tags-secondary"
                >
                  .less
                </code>
                 to
                 
                <code
                  className="code-tags code-tags-secondary"
                >
                  .css
                </code>
              </li>
              <li>
                e.g.
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/src/less/components/topbar.less
                </code>
                 
                or
                 
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/src/scripts/main/topbar.js
                </code>
              </li>
            </ul>
          </li>
          <li>
            bundled prefixed transpiled files
            <ul>
              <li>
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/dist/...
                </code>
              </li>
              <li>
                bundled
              </li>
              <li>
                <code
                  className="code-tags code-tags-secondary"
                >
                  .js
                </code>
                 &
                 
                <code
                  className="code-tags code-tags-secondary"
                >
                  .css
                </code>
              </li>
              <li>
                processed for browser support (CSS is auto-prefixed, JS is transpiled)
              </li>
              <li>
                e.g.
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/dist/designguide/styles/swedbankpay.css
                </code>
                 
                or
                 
                <code
                  className="code-tags code-tags-secondary"
                >
                  @swedbankpay/design-guide/dist/designguide/scripts/dg.js
                </code>
              </li>
            </ul>
          </li>
        </ul>
      </p>
    </section>
    <section>
      <h3>
        JavaScript
      </h3>
      <p>
        Some of the Design System components require javascript (not all of them). To achieve this:
        <ul>
          <li>
            Import the scripts of each component used
          </li>
          <li>
            Once imported, to initialize each component we need to call its \`init()\`method.
          </li>
        </ul>
      </p>
      <div
        className="alert alert-informative"
      >
        <i
          aria-hidden="true"
          className="at-info-circle"
        />
        <p>
          <b>
            Recommended way:
          </b>
           Importing component per component means you do not end up importing the javascript for components you are not using. This way your js served to the user is smaller, thus enhancing a faster experience for the users.
        </p>
      </div>
      <p>
        Importing component per component:
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            JAVASCRIPT
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                    2
                    

                    3
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    import { expandable, sheet } from "@swedbankpay/design-guide"; 
                    

                    expandable.init(); 
                    

                    sheet.init();
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <p>
        You can also import the whole bundle:
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            JAVASCRIPT
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                    2
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    import dg from "@swedbankpay/design-guide"; 
                    

                    dg.script.initAll();
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <div
        className="alert alert-informative"
      >
        <i
          aria-hidden="true"
          className="at-info-circle"
        />
        <p>
          <b>
            Standards:
          </b>
           Our JS uses modern features such as optional-chaining. If you're experiencing issues you might want to look at our browser support. In the case our browser support is not enough for your project you might want to use polyfills and/or Babel transpiling to fix it.
          <br />
          Also, some build tools such as Webpack 4 do not support some of those features. You can then use Babel to transpile the code from the Design System package.
        </p>
      </div>
    </section>
    <section>
      <h3>
        CSS/Less
      </h3>
      <p>
        For our CSS you can choose to import it from a
         
        <code
          className="code-tags code-tags-secondary"
        >
          css/less/scss
        </code>
         file, from a
         
        <code
          className="code-tags code-tags-secondary"
        >
          js
        </code>
         file if your bundler supports it,
         
        <em>
          or
        </em>
         point to the file directly from your bundler.
      </p>
      <p>
        We do not have a recommended way so far. This being said, importing from the style file might be easier and can leverage the CSS layers (
         
        <code
          className="code-tags code-tags-secondary"
        >
          @layer
        </code>
         ).
      </p>
      <h4>
        Importing from a style file
      </h4>
      <p>
        Example from a 
        <code
          className="code-tags code-tags-secondary"
        >
          .less
        </code>
         file
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            CSS
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    @import url("@swedbankpay/design-guide/src/less/TESTBRAND.less");
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <h4>
        Importing through the build tool
      </h4>
      <p>
        Example from 
        <code
          className="code-tags code-tags-secondary"
        >
          angular.json
        </code>
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            JSON
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                    2
                    

                    3
                    

                    4
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    "styles": [ 
                    

                    Â  "src/app.less", 
                    

                    Â  "./node_modules/@swedbankpay/design-guide/src/less/TESTBRAND.less" 
                    

                    ],
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <p>
        Example from 
        <code
          className="code-tags code-tags-secondary"
        >
          nuxt.config.js
        </code>
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            JAVASCRIPT
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                    2
                    

                    3
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    "css": [ 
                    

                    Â  "./node_modules/@swedbankpay/design-guide/src/less/TESTBRAND.less" 
                    

                    ],
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <div
        className="alert alert-informative"
      >
        <i
          aria-hidden="true"
          className="at-info-circle"
        />
        <p>
          <b>
            Import only once:
          </b>
           Either import from the style file 
          <em>
            OR
          </em>
           
          from the build tool. If you import through both you will end up loading twice the styles on the client, which has a negative impact on the web-performance.
        </p>
      </div>
    </section>
    <section>
      <h3>
        Installing individual components
      </h3>
      <p>
        Import 
        <code
          className="code-tags code-tags-primary"
        >
          variables-TESTBRAND.less
        </code>
         and the desired component, i.e. Expandable:
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            CSS
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                    2
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    @import url("@swedbankpay/design-guide/src/less/variables-TESTBRAND.less");
                    

                    @import url("@swedbankpay/design-guide/src/less/components/expandable.less");
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <div
        className="alert alert-informative"
      >
        <i
          aria-hidden="true"
          className="at-info-circle"
        />
        <p>
          <b>
            Web-perf:
          </b>
           Importing the style of individual components means fewer css imported, hence smaller file size. If you do not want to import them individually then you might want to check about tree-shaking (purging dead code).
        </p>
      </div>
    </section>
    <section>
      <h3>
        Design guide light demonstration
      </h3>
      <p>
        In this video we show a real life example on how to use the design guide through NPM.
      </p>
      <div
        className="video"
      >
        <iframe
          allowFullScreen={true}
          className="responsive-iframe"
          frameBorder="0"
          src="https://player.vimeo.com/video/727352249?h=232b0bec4c&amp;badge=0&amp;autopause=0&amp;player_id=0&amp;app_id=58479"
        />
      </div>
    </section>
  </section>
  <section>
    <h2
      id="installation"
    >
      Installation with CDN
    </h2>
    <p>
      To quickly add the 
      TESTBRAND
       Design Guide to your project, include the snippets below in your project.
    </p>
    <section>
      <h3>
        Include in the header
      </h3>
      <p>
        Copy-paste the following CSS code into
         
        <code
          className="code-tags code-tags-secondary"
        >
          &lt;head&gt;
        </code>
         before all the other stylesheets in order to load our CSS.
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            HTML
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    &lt;link rel="stylesheet" href="http://localhost/TEST/styles/TESTBRAND.css" /&gt;
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
    </section>
    <section>
      <h3>
        Include at the bottom
      </h3>
      <p>
        Many of our components requires the use of JavaScript in order to function. Place the following
         
        <code
          className="code-tags code-tags-secondary"
        >
          &lt;Script&gt;
        </code>
         tag near the end of your pages, right before the closing
         
        <code
          className="code-tags code-tags-secondary"
        >
          &lt;/body&gt;
        </code>
         tag to enable them.
      </p>
      <figure>
        <div
          className="code-view"
        >
          <header
            className="code-view-header"
          >
            HTML
            <button
              aria-describedby="tooltipCopy"
              className="tooltip"
              onClick={[Function]}
            >
              <i
                aria-hidden="true"
                className="at-clipboard small"
              />
              <div
                id="tooltipCopy"
                role="tooltip"
              >
                Copy to clipboard
              </div>
            </button>
          </header>
          <table
            className="code-view-table"
          >
            <tbody>
              <tr>
                <td
                  className="code-view-numbers"
                >
                  <pre>
                    1
                    

                  </pre>
                </td>
                <td
                  className="code-view-code"
                >
                  <pre>
                    &lt;script src="http://localhost/TEST/scripts/dg.js" global="true" autoload="true"&gt;&lt;/script&gt;
                    

                  </pre>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </figure>
      <p>
        The 
        <code
          className="code-tags code-tags-secondary"
        >
          global
        </code>
         attribute adds the script to the global scope. The
         
        <code
          className="code-tags code-tags-secondary"
        >
          autoload
        </code>
         attribute automatically initiates all components when loaded. Great for plug and play.
      </p>
    </section>
  </section>
  <section>
    <h3>
      How to initialize our JavaScript components
    </h3>
    <p>
      Our script runs 
      <code
        className="code-tags code-tags-secondary"
      >
        dg.script.initAll();
      </code>
       
      on DOMContentLoaded making it so you have to run
       
      <code
        className="code-tags code-tags-secondary"
      >
        dg.&lt;component_name&gt;.init();
      </code>
       manually when a component is rendered after the event has fired.
    </p>
    <p>
      To initialize components use
       
      <code
        className="code-tags code-tags-secondary"
      >
        dg.&lt;component_name&gt;.init();
      </code>
      . If the function is called without arguments it will target all components with class 
      <code
        className="code-tags code-tags-secondary"
      >
        &lt;component_name&gt;
      </code>
       on the page. Initializing only one component can be done by passing an ID, it must match an existing components ID and will print a warning message if the value is invalid.
    </p>
    <p>
      If only one component is initialized then the init function will return one object. If several components are initialized it will return an array of objects.
    </p>
    <figure>
      <div
        className="code-view"
      >
        <header
          className="code-view-header"
        >
          HTML
          <button
            aria-describedby="tooltipCopy"
            className="tooltip"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="at-clipboard small"
            />
            <div
              id="tooltipCopy"
              role="tooltip"
            >
              Copy to clipboard
            </div>
          </button>
        </header>
        <table
          className="code-view-table"
        >
          <tbody>
            <tr>
              <td
                className="code-view-numbers"
              >
                <pre>
                  1
                  

                  2
                  

                </pre>
              </td>
              <td
                className="code-view-code"
              >
                <pre>
                  dg.component.init();
                  

                  dg.component.init("demo-component-id");
                  

                </pre>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </figure>
  </section>
  <section>
    <h2
      id="browser-support"
    >
      Browser Support
    </h2>
    <p>
      The design guide is tested and currently supported by these browsers on the bundled files ( i.e. when importing from
      <code
        className="code-tags code-tags-secondary"
      >
        @swedbankpay/design-guide/dist/...
      </code>
       ).
    </p>
    <div
      className="browser-support"
    >
      <div>
        <img
          alt="Chrome logo"
          src="/TEST/img/documentation/forDevelopers/chrome.png"
        />
        <p>
          Yes
        </p>
      </div>
      <div>
        <img
          alt="Edge logo"
          src="/TEST/img/documentation/forDevelopers/edge.png"
        />
        <p>
          Yes
        </p>
      </div>
      <div>
        <img
          alt="Firefox logo"
          src="/TEST/img/documentation/forDevelopers/firefox.png"
        />
        <p>
          Yes
        </p>
      </div>
      <div>
        <img
          alt="Opera logo"
          src="/TEST/img/documentation/forDevelopers/opera.png"
        />
        <p>
          Yes
        </p>
      </div>
      <div>
        <img
          alt="Safari logo"
          src="/TEST/img/documentation/forDevelopers/safari.png"
        />
        <p>
          Yes
        </p>
      </div>
    </div>
    <div
      className="alert alert-informative mt-4"
    >
      <i
        aria-hidden="true"
        className="at-info-circle"
      />
      <p>
        <b>
          Target Browser:
        </b>
         "Last 3 versions, &gt; 1%, not dead"
      </p>
    </div>
    <h4>
      Browser support for products using the Design Guide
    </h4>
    <p>
      It is up to every Project manager and development team to decide which browsers their product should support. If you want to support other browsers than the ones supported by the Design Guide you need to maintain and test the component code in in your project.
       
    </p>
  </section>
</section>
`;
